README
NAME
    Dist::Zilla::Plugin::Conflicts - Declare conflicts for your distro

VERSION
    version 0.14

SYNOPSIS
    In your dist.ini:

      [Conflicts]
      Foo::Bar = 0.05
      Thing    = 2

DESCRIPTION
    This module lets you declare conflicts on other modules (usually
    dependencies of your module) in your dist.ini.

    Declaring conflicts does several thing to your distro.

    First, it generates a module named something like
    "Your::Distro::Conflicts". This module will use Dist::CheckConflicts to
    declare and check conflicts. The package name will be obscured from
    PAUSE by putting a newline after the "package" keyword.

    All of your runtime prereqs will be passed in the "-also" parameter to
    Dist::CheckConflicts.

    Second, it adds code to your Makefile.PL or Build.PL to load the
    generated module and print warnings if conflicts are detected.

    Finally, it adds the conflicts to the META.json and/or META.yml files
    under the "x_breaks" key.

USAGE
    Using this module is simple, add a "[Conflicts]" section and list each
    module you conflict with:

      [Conflicts]
      Module::X = 0.02

    The version listed is the last version that *doesn't* work. In other
    words, any version of "Module::X" greater than 0.02 should work with
    this release.

    The special key "-script" can also be set, and given the name of a
    script to generate, as in:

      [Conflicts]
      -script   = bin/foo-conflicts
      Module::X = 0.02

    This script will be installed with your module, and can be run to check
    for currently installed modules which conflict with your module. This
    allows users an easy way to fix their conflicts - simply run a command
    such as "foo-conflicts | cpanm" to bring all of your conflicting modules
    up to date.

    Note: Currently, this plugin only works properly if it is listed in your
    dist.ini *after* the plugin which generates your Makefile.PL or
    Build.PL. This is a limitation of Dist::Zilla that will hopefully be
    addressed in a future release.

SEE ALSO
    *   Dist::CheckConflicts

    *   Dist::Zilla::Plugin::Breaks

    *   Dist::Zilla::Plugin::Test::CheckBreaks

SUPPORT
    Please report any bugs or feature requests to
    "bug-dist-zilla-plugin-conflicts@rt.cpan.org", or through the web
    interface at <http://rt.cpan.org>. I will be notified, and then you'll
    automatically be notified of progress on your bug as I make changes.

DONATIONS
    If you'd like to thank me for the work I've done on this module, please
    consider making a "donation" to me via PayPal. I spend a lot of free
    time creating free software, and would appreciate any support you'd care
    to offer.

    Please note that I am not suggesting that you must do this in order for
    me to continue working on this particular software. I will continue to
    do so, inasmuch as I have in the past, for as long as it interests me.

    Similarly, a donation made in this way will probably not make me work on
    this software much more, unless I get so many donations that I can
    consider working on free software full time, which seems unlikely at
    best.

    To donate, log into PayPal and send money to autarch@urth.org or use the
    button on this page: <http://www.urth.org/~autarch/fs-donation.html>

AUTHOR
    Dave Rolsky <autarch@urth.org>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2014 by Dave Rolsky.

    This is free software, licensed under:

      The Artistic License 2.0 (GPL Compatible)



Changes
0.17    2014-12-21

- Removed the use of autobox, following Dist::Zilla's lead. Patch by Karen
  Etheridge. GitHub PR #4.


0.16    2014-09-03

- Add a hint for [SurgicalPodWeaver] so it does not try to modify the file (no
  longer possible since version 0.15). Patch by Karen Etheridge. GitHub PR #3.

- Repository moved to the github moose organization.


0.15    2014-08-31

- Generate the Conflicts module as a FromCode file, so its content is not
  calculated until after prereqs are available: restores the -also section
  lost in version 0.09. Patch by Karen Etheridge. GitHub PR #2.

- The generated files now include comments indicating what version of this
  plugin was used to generate them.  Patch by Karen Etheridge. GitHub PR #1.


0.14    2014-04-06

- Same as 0.13001


0.13001 2014-04-06 (TRIAL RELEASE)

- Fixed the x_breaks metadata to properly indicate the problematic version
  range. Patch by Karen Etheridge. RT #94467.


0.13    2013-08-25

- MetaCPAN was picking up the "# ABSTRACT" line in the conflicts module
  template.

- Removed some incorrect docs left over from 0.11


0.12    2013-08-25

- Fix bad conflicts entries added to metaspec (Karen is very sorry!), broken
  in v0.11. Sort of resolves RT #87123.


0.11    2013-07-22

- This plugin now adds conflicts type entries to the distro's Meta
  information, per the CPAN meta spec. Patch by Karen Etheridge. RT #87123.


0.10    2013-06-07

- The x_conflicts key added to the META.* files has now been renamed to
  x_breaks, per the Lancaster Consensus of 2013
  (https://github.com/Perl-Toolchain-Gang/toolchain-site/blob/master/lancaster-consensus.md).

- The generated check_conflicts sub added to the Makefile.PL or Build.PL file
  now checks the AUTOMATED_TESTING and NONINTERACTIVE_TESTING env vars to
  determine whether or not it should sleep for a few seconds so its message
  can be read. These env vars also come from the Lancaster Consensus.


0.09    2013-03-31

- If the plugin is run before a Makefile.PL or Build.PL can be generated, a
  fatal error is thrown. RT#80271. (Karen Etheridge)

- The files this plugin generates are now added early enough to be seen by any
  FileMunger plugins. However, the PkgVersion plugin still skips the generated
  *::Conflicts module because the package name of the generated module is
  intentionally hidden from PAUSE.

- Added an ABSTRACT to the generated module.


0.08    2011-07-22

- Add Dist::CheckConflicts to both configures and runtime dependency lists for
  modules which use this plugin.


0.07    2011-01-27

- Fix module name in pod.

- Add a dep on Dist::CheckConflicts 0.02 to modules which use this plugin (not
  0.01).


0.06    2011-01-16

- The dist.ini wouldn't load if a -script wasn't defined in the Conflicts
  config section.


0.05    2011-01-03

- The conflict warning added to your Makefile.PL or Build.PL now mentions the
  generated script if you ask for one to be generated. (Jesse Luehrs)

- No -also will be passed to Dist::CheckConflicts if you don't have
  dependencies. (Jesse Luehrs)

- Require Dist::CheckConflicts 0.02, which has better -also handling.


0.04    2010-12-31

- In the generated conflict module, all of the distro's runtime prereqs are
  passed in the -also parameter to Dist::CheckConflicts.


0.03    2010-12-31

- Added support for a -script config key, which tells this plugin to generate
  a script of the given name that lists conflicts. (Jesse Luehrs)


0.02    2010-11-22

- This module loaded Test::ConsistentVersion, which it wasn't actually using.


0.01    2010-11-22

- Initial release


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.029.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Declare conflicts for your distro",
  "AUTHOR" => "Dave Rolsky <autarch\@urth.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Conflicts",
  "EXE_FILES" => [],
  "LICENSE" => "artistic_2",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::Conflicts",
  "PREREQ_PM" => {
    "Dist::CheckConflicts" => "0.02",
    "Dist::Zilla" => "4.0",
    "Dist::Zilla::File::FromCode" => 0,
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::InstallTool" => 0,
    "Dist::Zilla::Role::MetaProvider" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Differences" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "0.88",
    "Test::Requires" => 0,
    "if" => 0
  },
  "VERSION" => "0.17",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::CheckConflicts" => "0.02",
  "Dist::Zilla" => "4.0",
  "Dist::Zilla::File::FromCode" => 0,
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::InstallTool" => 0,
  "Dist::Zilla::Role::MetaProvider" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Differences" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "0.88",
  "Test::Requires" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



